src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -437,56 +429,48 @@ void SyncInferRequest::set_tensor(const ov::Output<const ov::Node>& in_port, con
         }
 
         if (!isDynamic && ov::shape_size(shape.to_shape()) != tensor->get_size()) {
-            OPENVINO_THROW("Can't set the output tensor with name: ",
-                           name,
+            OPENVINO_THROW("Can't set the output tensor with index: ",
+                           output_index,
                            ", because the model output size = ",
                            ov::shape_size(shape.to_shape()),
                            " and the currernt tensor size = ",
                            tensor->get_size(),
                            " are different.");
         }
 
-        const auto& desc = m_graph->getOutputNodeByName(name)->getParentEdgeAt(0)->getMemory().getDesc();
+        const auto& desc = m_graph->getOutputNodeByIndex(output_index)->getParentEdgeAt(0)->getMemory().getDesc();
         if (!isDynamic && mem_desc_ptr->isCompatible(desc)) {
-            m_external_ptr[name] = tensor;
-        } else if (m_external_ptr.find(name) != m_external_ptr.end()) {
-            m_external_ptr.erase(name);
+            m_output_external_ptr[output_index] = tensor;
+        } else if (m_output_external_ptr.find(output_index) != m_output_external_ptr.end()) {
+            m_output_external_ptr.erase(output_index);
         }
 
-        m_outputs[name] = tensor;
-        m_outputControlBlocks.erase(name); // now the memory is under user's control
+        m_outputs[output_index] = tensor;
+        m_outputControlBlocks.erase(output_index); // now the memory is under user's control
     }
     ov::ISyncInferRequest::set_tensor(port, tensor);
 }
 
 void SyncInferRequest::set_tensors_impl(const ov::Output<const ov::Node> port, const std::vector<ov::SoPtr<ITensor>>& tensors) {
-    for (const auto& input : get_inputs()) {
-        if (input == port) {
-            m_batched_tensors[input.get_tensor_ptr()] = tensors;
-            return;
-        }
+    if (find_port(port).is_input()) {
+        m_batched_tensors[port.get_tensor_ptr()] = tensors;
```

Comment:
I don't think that address of tensor is a reliable key.
We can have 2 different ports: ov::Model and ov::CompiledModel and their `get_tensor_ptr()` are different

Reply:
@ilya-lavrenov   m_batched_tensors defined in class ISyncInferRequest, which used in cpu, gpu, template plugin with get_tensor_ptr(), maybe we can get another reliable key in next PR only for this topic.
BTW, this is used in infer request, at this situation all ports are from ov::CompiledModel I think.

Reply:
@ilya-lavrenov yes, the same port from `ov::Model` and `ov::CompiledModel` should have the different `tensor_desc_ptr`, but using `tensor_desc_ptr` as key is located from inference base class to all plugins implement class, so it will need more modification if we want to implement one better solution to replace it. I agree with @zhaixuejun1993 to create a dedicated PR to solve it if we do need this improvement.

Reply:
ok